lxd/images.go

```diff
@@ -353,18 +381,15 @@ func imageDelete(d *Daemon, r *http.Request) Response {
 	return EmptySyncResponse
 }
 
-func imageGet(d *Daemon, r *http.Request) Response {
-	fingerprint := mux.Vars(r)["fingerprint"]
-
-	public := !d.isTrustedClient(r)
+func doImageGet(d *Daemon, fingerprint string, public bool) (shared.ImageInfo, Response) {
 	imgInfo, err := dbImageGet(d, fingerprint, public)
 	if err != nil {
-		return SmartError(err)
+		return shared.ImageInfo{}, SmartError(err)
```

Comment:
Oh, I certainly tried that, but Go wouldn't let me :)

../../go/src/github.com/lxc/lxd/lxd/images.go:392: cannot use nil as type shared.ImageInfo in return argument